src/java.desktop/share/classes/javax/swing/text/html/CSS.java

```diff
@@ -2965,6 +2965,17 @@ ImageIcon getImage(URL base) {
             }
             return image;
         }
+
+        @Override
+        public int hashCode() {
+            return Objects.hashCode(this.svalue);
+        }
+
+        @Override
+        public boolean equals(Object val) {
+            return val instanceof CSS.BackgroundImage img
+                   && Objects.equals(this.svalue, img.svalue);
```

Comment:
```suggestion
            return val instanceof CSS.BackgroundImage img
                   && Objects.equals(svalue, img.svalue);
```
Using `this` is redundant. You didn't use it in other implementations.